frame/staking/src/tests.rs

```diff
@@ -3317,14 +3331,126 @@ fn payout_stakers_handles_basic_errors() {
 		}
 		// We are at era 99, with history depth of 84
 		// We should be able to payout era 15 through 98 (84 total eras), but not 14 or 99.
-		assert_noop!(Staking::payout_stakers(Origin::signed(1337), 11, 14), Error::<Test>::InvalidEraToReward);
-		assert_noop!(Staking::payout_stakers(Origin::signed(1337), 11, 99), Error::<Test>::InvalidEraToReward);
+		assert_noop!(
+			Staking::payout_stakers(Origin::signed(1337), 11, 14),
+			Error::<Test>::InvalidEraToReward.with_weight(err_weight)
+		);
+		assert_noop!(
+			Staking::payout_stakers(Origin::signed(1337), 11, 99),
+			Error::<Test>::InvalidEraToReward.with_weight(err_weight)
+		);
 		assert_ok!(Staking::payout_stakers(Origin::signed(1337), 11, 15));
 		assert_ok!(Staking::payout_stakers(Origin::signed(1337), 11, 98));
 
 		// Can't claim again
-		assert_noop!(Staking::payout_stakers(Origin::signed(1337), 11, 15), Error::<Test>::AlreadyClaimed);
-		assert_noop!(Staking::payout_stakers(Origin::signed(1337), 11, 98), Error::<Test>::AlreadyClaimed);
+		assert_noop!(
+			Staking::payout_stakers(Origin::signed(1337), 11, 15),
+			Error::<Test>::AlreadyClaimed.with_weight(err_weight)
+		);
+		assert_noop!(
+			Staking::payout_stakers(Origin::signed(1337), 11, 98),
+			Error::<Test>::AlreadyClaimed.with_weight(err_weight)
+		);
+	});
+}
+
+#[test]
+fn payout_stakers_handles_weight_refund() {
+	// N.B. this test relies on the assumption that `payout_stakers_alive_staked` is solely used to
```

Comment:
What is `N.B.` and what else would be a valid assumption than this statement?

Reply:
[N.B. def here](https://en.wikipedia.org/wiki/Nota_bene#:~:text=Often%20abbreviated%20as%20NB%2C%20n.b.,of%20the%20subject%20being%20addressed.) - I interpret it as "note importantly", or "please pay close attention to this note"; I picked up from reading other code. I will change to `Note:` to reduce ambiguity.

From my understanding `payout_stakers_alive_staked` is the weight for a payout op to `RewardDestination::Staked`, the worst case for a payout op.

If we change this in the future to have weights specific to each reward destination variant (i.e. `payout_stakers_alive_{stash, controller, account}`) this assumption would no longer be valid.